.github/workflows/test.yaml

```diff
@@ -172,6 +172,71 @@ jobs:
           rm -rf /tmp/.buildx-cache
           mv /tmp/.buildx-cache-new /tmp/.buildx-cache
 
+  nx-docker-build:
+    # Only run on giraffe-develop branch for now
+    if: github.ref == 'refs/heads/giraffe-develop' || 'refs/heads/task/nx-deploy'
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./bciers
+    needs: backend-docker-build
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Docker metadata
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: ghcr.io/bcgov/cas-rep-frontend
+          tags: |
+            type=sha,format=long,prefix=
+            latest
+            type=ref,event=pr
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/setup-node@v3
+        with:
+          cache: 'yarn'
+      - name: Install dependencies
+        run: yarn install --frozen-lockfile
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v4
+        with:
+          # This will need to be updated when giraffe-develop merges with develop
+          main-branch-name: "giraffe-develop"
+      - name: Cache Docker layers
+        uses: actions/cache@v3
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-client-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-client
+      - name: Build images
+        env:
+          INPUT_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          INPUT_PUSH: true
+          INPUT_TAGS: ${{ steps.meta.outputs.tags }}
+          INPUT_LABELS: ${{ steps.meta.output.labels }}
+          INPUT_CACHE_FROM: type=local,src=/tmp/.buildx-cache
+          INPUT_CACHE_TO: type=local,dest=/tmp/.buildx-cache-new
+        run: |
+          npx nx container reporting --skip-nx-cache
+        # Temp fix
+        # https://github.com/docker/build-push-action/issues/252
+        # https://github.com/moby/buildkit/issues/1896
+      - name: Move cache
+        run: |
+          rm -rf /tmp/.buildx-cache
+          mv /tmp/.buildx-cache-new /tmp/.buildx-cache
+
```

Comment:
This code will need to be addressed, as it is flagged to only run on merge to `giraffe-develop`. The makefiles/helm templates will be what actually pulls the image for deploy, so it should be fine.